### ISSUE TITLE
Severely breaks the Android folder picker

### ISSUE THREAD

**Issue by derivativeoflog7** [DanGLES3/NoStorageRestrict#8]
The module severely breaks the folder picker. Folders like the root of the internal storage and Downloads (all the folders that the module touches?) can't be accessed anymore showing the error in the screenshot below (can't load content at the moment).
Additionally, it breaks things in such a way that disabling the module doesn't suffice, I have to dirty flash my ROM again to fix it.
Rom is evolution_umi-userdebug 13 TQ1A.230105.001.A2 1673888029

![Screenshot_20230312-194659_Files.png](https://user-images.githubusercontent.com/100133857/224567542-35971980-7cad-4219-8bd2-863db1fccfa1.png)

**Comment by derivativeoflog7** [comment]
Fortunately the xposed version doesn't seem to break anything, so I'll use that instead (since I really really really really really can't stand this stupid restriction on the folders)

**Comment by DanGLVK** [comment]
> Fortunately the xposed version doesn't seem to break anything, so I'll use that instead (since I really really really really really can't stand this stupid restriction on the folders)

Yeah, tbh it's hard to make the Magisk version reliable due to it doing APK replacement of an system app, some devices/ROMs just refuse to accept it

That's why it's best to use the Xposed version as it just patches the needed code in.

**Comment by tg5hKVHPyX** [comment]
Is there any way to fix this? I installed the magisk module because I wasn't aware that it replaces the system app. I'm using samsung a70 on android 10 stock rom. I extracted the original apk from the stock rom firmware files, but trying to install it with adb doesn't work.

**Comment by derivativeoflog7** [comment]
> Is there any way to fix this? I installed the magisk module because I wasn't aware that it replaces the system app. I'm using samsung a70 on android 10 stock rom. I extracted the original apk from the stock rom firmware files, but trying to install it with adb doesn't work.

When it happened to me, the quickest way to fix it was dirty flashing the ROM again (without wiping any data) (remembering to reinstall Magisk)

**Comment by tg5hKVHPyX** [comment]
> > Is there any way to fix this? I installed the magisk module because I wasn't aware that it replaces the system app. I'm using samsung a70 on android 10 stock rom. I extracted the original apk from the stock rom firmware files, but trying to install it with adb doesn't work.
> 
> When it happened to me, the quickest way to fix it was dirty flashing the ROM again (without wiping any data) (remembering to reinstall Magisk)

Do you mean flashing home_csc?

**Comment by DerivativeOfLog7** [comment]
I don't know how it works on Odin since Samsung is the special kid that decided to handle stock ROMs and flashing them in their own unique way... In my case it was flashing the .zip from TWRP, which is not your case

**Comment by derivativeoflog7** [comment]
> > > Is there any way to fix this? I installed the magisk module because I wasn't aware that it replaces the system app. I'm using samsung a70 on android 10 stock rom. I extracted the original apk from the stock rom firmware files, but trying to install it with adb doesn't work.
> > 
> > 
> > When it happened to me, the quickest way to fix it was dirty flashing the ROM again (without wiping any data) (remembering to reinstall Magisk)
> 
> Do you mean flashing home_csc?

I don't know how it works on Odin since Samsung is the special kid that decided to handle stock ROMs and flashing them in their own unique way... In my case it was flashing the .zip from TWRP, which is not your case

**Comment by DanGLES3** [comment]
> 

Odd,, it only replaces it while the module is active, it shouldn't break after the module was removed

**Comment by DanGLVK** [comment]
> Is there any way to fix this? I installed the magisk module because I wasn't aware that it replaces the system app. I'm using samsung a70 on android 10 stock rom. I extracted the original apk from the stock rom firmware files, but trying to install it with adb doesn't work.

Odd, it only replaces it while the module is active, it shouldn't break after the module was removed,, I'll add an warning against using the Magisk version on stock ROMs and recommend the Lsposed version instead 

Sorry for everything

**Comment by derivativeoflog7** [comment]
> > Is there any way to fix this? I installed the magisk module because I wasn't aware that it replaces the system app. I'm using samsung a70 on android 10 stock rom. I extracted the original apk from the stock rom firmware files, but trying to install it with adb doesn't work.
> 
> Odd, it only replaces it while the module is active, it shouldn't break after the module was removed,, I'll add an warning against using the Magisk version on stock ROMs and recommend the Lsposed version instead 
> 
> Sorry for everything 

I'm pretty sure that's a Magisk issue, since it also happened to me with other modules. Apparently disabling or even deleting a module doesn't remove everything. (I'm on an aosp based custom rom)

**Comment by derivativeoflog7** [comment]
But also, this same custom rom (which is Android 13 based) has the issue I described in the OP

**Comment by DanGLVK** [comment]
> But also, this same custom rom (which is Android 13 based) has the issue I described in the OP

Tbh I've never been a fan of the Magisk version of this module, it's not rare that one ROM or another or another will throw a tantrum and break because an app was replaced

Xposed just does the job while being much less intrusive
